client/src/components/App.js

```diff
@@ -21,118 +21,86 @@ import { Layout } from "antd";
 import "antd/dist/antd.css";
 const { Footer } = Layout;
 
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { user: {}, loginAttention: false };
-  }
+export default function App(props) {
+  const [user, setUser] = useState({});
+  const [loginAttention, setLoginAttention] = useState(false);
 
-  componentDidMount() {
+  useEffect(() => {
     get("/api/whoami").then((res) => {
-      this.setState({ user: res });
+      setUser(res);
     });
-  }
+  }, []);
 
-  updateUser = (user) => {
-    this.setState({ user });
+  const updateUser = (user) => {
+    setUser({ user });
   };
 
-  setLoginAttention = (value) => {
-    this.setState({ loginAttention: value });
+  const toggleLoginAttention = (value) => {
```

Comment:
i dont think we need `updateUser` or `toggleLoginAttention` anymore, since they're equivalent to `setUser` and `setLoginAttention` respectively

can we delete these functions, and directly pass `setUser` and `setLoginAttention` to the components below?

Reply:
Oki, understood.

Reply:
Currently has errors, trying to aswell convert affected components now to Hooks.

Reply:
o did i merge too soon LOL